hypothesis_geometry/planar.py

```diff
@@ -162,12 +167,82 @@ def triangular_contours(x_coordinates: Strategy[Coordinate],
             .map(list))
 
 
+def segments(x_coordinates: Strategy[Coordinate],
+             y_coordinates: Optional[Strategy[Coordinate]] = None
+             ) -> Strategy[Segment]:
+    """
+    Returns a strategy for segments.
+
+    :param x_coordinates: strategy for endpoints' x-coordinates.
+    :param y_coordinates:
+        strategy for endpoints' y-coordinates,
+        ``None`` for reusing x-coordinates strategy.
+    """
+
+    def non_degenerate_segment(segment: Segment) -> bool:
+        start, end = segment
+        return start != end
+
+    points_strategy = points(x_coordinates, y_coordinates)
+    return (strategies.tuples(points_strategy, points_strategy)
+            .filter(non_degenerate_segment))
+
+
+def polylines(x_coordinates: Strategy[Coordinate],
+              y_coordinates: Optional[Strategy[Coordinate]] = None,
+              *,
+              min_size: int = MIN_POLYLINE_SIZE,
+              max_size: Optional[int] = None) -> Strategy[Polyline]:
+    """
+    Returns a strategy for polylines.
```

Comment:
You might add information on which types of polylines this can currently generate.

Reply:
I guess it can potentially generate any kind of them, so no need to specify